src/vdoc/settings/__init__.py

```diff
@@ -26,3 +29,37 @@ class VDocSettings(BaseSettings):
     bind_port: int = DEFAULT_BIND_PORT
 
     project_display_name_mapping: dict[str, str] = {}
+
+    project_categories: list[ProjectCategory] = []
+    project_category_mapping: dict[str, str] = {}
+
+    @model_validator(mode="after")
+    def validate_model(self) -> Self:
+        """Validates the model.
+
+        The following checks are performed:
+
+        - Ensures that all category IDs in `project_categories` are unique.
+        - Ensures that all category names in `project_categories` are unique.
+        - Ensures that all categories in `project_category_mapping` are defined in `project_categories`.
+
+        Raises:
+            ValueError: If any check fails.
+
+        Returns:
+            Self: The validated model.
+        """
+        project_category_ids = list(map(lambda category: category.id, self.project_categories))
+        project_category_names = list(map(lambda category: category.name, self.project_categories))
+        if len(set(project_category_ids)) != len(project_category_ids):
+            raise ValueError("Duplicate category IDs are not allowed in `project_categories`")
+        if len(set(project_category_names)) != len(project_category_names):
+            raise ValueError("Duplicate category names are not allowed in `project_categories`")
+        for category_name in self.project_category_mapping.values():
+            if category_name not in map(lambda category: category.name, self.project_categories):
```

Comment:
```suggestion
            if category_name not in project_category_names:
```
Die Namen sind doch schon in einer Liste, hier wird unnötig rumgerechnet